tgui/packages/tgui/interfaces/ChemMaster.js

```diff
@@ -1,431 +1,450 @@
+                {result.blood_dna}
+              </LabeledList.Item>
+            </Fragment>
+          )}
+          {!data.condi && (
+            <Button
+              icon={data.printing ? 'spinner' : 'print'}
+              disabled={data.printing}
+              iconSpin={!!data.printing}
+              ml="0.5rem"
+              content="Print"
+              onClick={() => act('print', {
+                idx: result.idx,
+                beaker: modal.args.beaker,
+              })}
+            />
+          )}
+        </LabeledList>
+      </Box>
+    </Section>
+  );
+};
 
 export const ChemMaster = (props, context) => {
   const { data } = useBackend(context);
-  const { screen } = data;
+  const {
+    condi,
+    beaker,
+    beaker_reagents = [],
+    buffer_reagents = [],
+    mode,
+  } = data;
   return (
     <Window
-      width={465}
-      height={550}
+      width={575}
+      height={500}
       resizable>
-      <Window.Content scrollable>
-        {screen === 'analyze' && (
-          <AnalysisResults />
-        ) || (
-          <ChemMasterContent />
-        )}
+      <ComplexModal />
+      <Window.Content scrollable className="Layout__content--flexColumn">
+        <ChemMasterBeaker
+          beaker={beaker}
+          beakerReagents={beaker_reagents}
+          bufferNonEmpty={buffer_reagents.length > 0}
+        />
+        <ChemMasterBuffer
+          mode={mode}
+          bufferReagents={buffer_reagents}
+        />
+        <ChemMasterProduction
+          isCondiment={condi}
+          bufferNonEmpty={buffer_reagents.length > 0}
+        />
+        {/* <ChemMasterCustomization /> */}
       </Window.Content>
     </Window>
   );
 };
 
-const ChemMasterContent = (props, context) => {
+const ChemMasterBeaker = (props, context) => {
   const { act, data } = useBackend(context);
   const {
-    screen,
-    beakerContents = [],
-    bufferContents = [],
-    beakerCurrentVolume,
-    beakerMaxVolume,
-    isBeakerLoaded,
-    isPillBottleLoaded,
-    pillBottleCurrentAmount,
-    pillBottleMaxAmount,
-  } = data;
-  if (screen === 'analyze') {
-    return <AnalysisResults />;
-  }
+    beaker,
+    beakerReagents,
+    bufferNonEmpty,
+  } = props;
   return (
-    <Fragment>
-      <Section
-        title="Beaker"
-        buttons={!!data.isBeakerLoaded && (
-          <Fragment>
-            <Box inline color="label" mr={2}>
-              <AnimatedNumber
-                value={beakerCurrentVolume}
-                initial={0} />
-              {` / ${beakerMaxVolume} units`}
+    <Section
+      title="Beaker"
+      buttons={bufferNonEmpty
+        ? <Button.Confirm
+          icon="eject"
+          disabled={!beaker}
+          content="Eject and Clear Buffer"
+          onClick={() => act('eject')} />
+        : <Button
+          icon="eject"
+          disabled={!beaker}
+          content="Eject and Clear Buffer"
+          onClick={() => act('eject')} />}>
```

Comment:
holy shit who coded this
```suggestion
    <Section
      title="Beaker"
      buttons={bufferNonEmpty
        ? (
          <Button.Confirm
            icon="eject"
            disabled={!beaker}
            content="Eject and Clear Buffer"
            onClick={() => act('eject')} />
        ) : (
          <Button
            icon="eject"
            disabled={!beaker}
            content="Eject and Clear Buffer"
            onClick={() => act('eject')} />
        )}>
```